src/transaction-runner.ts

```diff
@@ -202,12 +209,11 @@ export abstract class Runner<T> {
     const transaction = this.session.transaction(
       (this.session.parent as Database).queryOptions_
     );
-    if (this.options.optimisticLock) {
-      transaction.useOptimisticLock();
-    }
-    if (this.options.excludeTxnFromChangeStreams) {
-      transaction.excludeTxnFromChangeStreams();
-    }
+    transaction!.setReadWriteTransactionOptions(
+      this.options && Object.keys(this.options).length
+        ? this.options
+        : this.session.parent._getSpanner().defaultTransactionOptions
```

Comment:
Do not use _getSpanner() private method . Instead create a private method `_getSpanner` in this class and use it. Refer this method in all other classes

Reply:
have refactored the code, please check